tests/test_exception.py

```diff
@@ -7,11 +9,28 @@ class MyError(Exception):
     """A sample exception for use by the tests in this module."""
 
 
+class NotMyError(Exception):
+    """A sample exception for use by the tests in this module."""
+
+
 def raise_my_error():
     """Raise ``MyError``."""
     raise MyError()
 
 
+def raise_not_my_error():
+    """Raise ``MyError``."""
+    raise NotMyError()
+
+
+error_gen = cycle((MyError(), NotMyError()))
```

Comment:
This is clever, but consider just using parametrization to pass the error class to raise through to `raise_my_error_or_not_my_error`, so that unit test coverage and failures are more transparent.

Reply:
I step into this again, please comment new solution